dockerhub_config_update.sh

```diff
@@ -0,0 +1,119 @@
+# 4. Run this command
+#
+set -o pipefail
+
+source ./common_functions.sh
+
+latest_version="12"
+hotspot_latest_tags="hotspot, latest"
+openj9_latest_tags="openj9"
+
+# Get the latest git commit of the current repo.
+# This is assumed to have all the latest dockerfiles already.
+gitcommit=$(git log | head -1 | awk '{ print $2 }')
+
+echo "# AdoptOpenJDK official images for OpenJDK with HotSpot and OpenJDK with Eclipse OpenJ9."
+echo
+echo "Maintainers: Dinakar Guniguntala <dinakar.g@in.ibm.com> (@dinogun)"
+echo "GitRepo: https://github.com/AdoptOpenJDK/openjdk-docker.git"
+
+# Iterate through all the VMs, for each supported version and packages to
+# generate the config file.
+for vm in ${all_jvms}
+do
+	for ver in ${supported_versions}
+	do
+		echo
+		echo "#------------------------------${vm} v${ver} images-------------------------------"
+		for pkg in ${all_packages}
+		do
+			echo
+			# Iterate through each of the Dockerfiles.
+			for file in $(find . -name "Dockerfile.*" | grep "/${ver}" | grep "${vm}" | grep "${pkg}")
+			do
+				# file will look like ./12/jdk/debian/Dockerfile.openj9.nightly.slim
+				# dockerfile name
+				dfname=$(basename ${file})
+				# dockerfile dir
+				dfdir=$(dirname ${file} | cut -c 3-)
+				os=$(echo ${file} | awk -F '/' '{ print $4 }')
+				# build = release or nightly
+				build=$(echo ${dfname} | awk -F "." '{ print $3 }')
+				# btype = full or slim
+				btype=$(echo ${dfname} | awk -F "." '{ print $4 }')
+
+				# Currently we are not pushing docker images for Alpine, Debian and Windows
+				if [ "${os}" == "windows" -o "${os}" == "alpine" -o "${os}" == "debian" ]; then
+					continue;
+				fi
+
+				# We do not push our nightly and slim images either.
+				if [ "${build}" == "nightly" -o "${btype}" == "slim" ]; then
+					continue;
+				fi
+
+				# Generate the tags
+				full_version=$(grep "VERSION" ${file} | awk '{ print $3 }')
+				ojdk_version=$(echo ${full_version} | sed 's/\(jdk\|jdk-\)//' | awk -F '_' '{ print $1 }')
+				ojdk_version=$(echo ${ojdk_version} | sed 's/+/_/')
+
+				full_ver_tag="${ojdk_version}-${pkg}"
+				# Add the openj9 version
+				if [ "${vm}" == "openj9" ]; then
+					openj9_version=$(echo ${full_version} | awk -F '_' '{ print $2 }')
+					full_ver_tag="${full_ver_tag}-${openj9_version}"
+				else
+					full_ver_tag="${full_ver_tag}-${vm}"
+				fi
+				ver_tag="${ver}-${pkg}-${vm}"
+				all_tags="${full_ver_tag}, ${ver_tag}"
+				if [ "${pkg}" == "jdk" ]; then
+					jdk_tag="${ver}-${vm}"
+					all_tags="${all_tags}, ${jdk_tag}"
+					# Add the "latest", "hotspot" and "openj9" tags for the right version
+					if [ "${ver}" == "${latest_version}" ]; then
+						vm_tags="${vm}_latest_tags"
+						eval vm_tags_val=\${$vm_tags}
+						all_tags="${all_tags}, ${vm_tags_val}"
+					fi
+				fi
+
+				# Generate the supported arches for the above tags.
+				arches=$(echo $(grep ') \\' ${file} | \
+					sed 's/\(ppc64el\|x86_64\|aarch64\)//g' | \
```

Comment:
Again should these values be held in an 'array' further up?

Reply:
Added comments to make it clear as an array doesn't make sense for this.